src/SparseMatrix.java

```diff
@@ -0,0 +1,322 @@
+public class SparseMatrix
```

Comment:
"Object should be immutable" for this you have to make it final.
For ref: http://www.javapractices.com/topic/TopicAction.do?Id=29

Reply:
done